### DISCUSSION TITLE
0.4.1d

### DISCUSSION THREAD

**Discussion by alienatedsec** [alienatedsec/yi-hack-v5#317] [Announcements]
## What's Changed in `0.4.1d`
Changes by @alienatedsec 
* Fixing cloudAPI workflow for CN based cameras
* Removed duplicate configs

**Full Changelog**: https://github.com/alienatedsec/yi-hack-v5/compare/0.4.0c...0.4.1d

## What's Changed in `0.4.1c`
Changes by @alienatedsec 
* `wpa_supplicant` upgrade to `v2.10` - this is to prevent boot loop issues experienced with certain Wi-Fi configurations (e.g., multiple APs with the same SSID) or any modern Wi-Fi networks (e.g., mixed mode WPA/WPA2 - subject to be removed in the future `wpa_supplicant` releases)
* Should fix #312 and the boot loop issues discussed in the #291  
* Support for WPA3-Personal - Simultaneous Authentication of Equals (SAE)
* Support to encrypt your Wi-Fi password with `wpa_passphrase` - not tested as of yet

**Full Changelog**: https://github.com/alienatedsec/yi-hack-v5/compare/0.4.0b...0.4.1c

Follow with discussions here: https://github.com/alienatedsec/yi-hack-v5/discussions/316

## What's Changed in `0.4.1b`
* Merging tested commits by @alienatedsec in https://github.com/alienatedsec/yi-hack-v5/pull/299

Changes by @roleoroleo 
- [Change size of fifo queues](https://github.com/alienatedsec/yi-hack-v5/pull/299/commits/ee0da9d0eb8afdb560953dca9d7f9ac2631d8f92)
- [Restore audio stream ch0_2.h264](https://github.com/alienatedsec/yi-hack-v5/pull/299/commits/66030f5abbbd86f186907809cb51c0a3a0af898f)
- [Add native api to take low res snapshot](https://github.com/alienatedsec/yi-hack-v5/pull/299/commits/cf13502b466fd084cfa7d442ca128056ab7540e8)
- [Remove option: watermark for onvif snapshot](https://github.com/alienatedsec/yi-hack-v5/pull/299/commits/15e07ff6e9e2d5d39af4fb8da58cb7c325cf32cb)
- [Change label for "Force low res snapshot"](https://github.com/alienatedsec/yi-hack-v5/pull/299/commits/c821423f61da34d030435b4081c5290e9221b5f4)
- [Improve RTSP server](https://github.com/alienatedsec/yi-hack-v5/pull/299/commits/54b1e653d8aa6136537568b3d8a4b8d7b66bb7cb)
- [Minor fixes](https://github.com/alienatedsec/yi-hack-v5/pull/299/commits/e6ebbce764ecd559ec5d612619caba1f9197b154)
- [Remove unused options when empty password](https://github.com/alienatedsec/yi-hack-v5/pull/299/commits/9c4fff63880b10779c4a4802f22503fb29cd61d6)
- [Add mqttv4 to watchdog](https://github.com/alienatedsec/yi-hack-v5/pull/299/commits/434db6d2ac1c70da9d7895087fcbf48f7fe562a5)
- [Add "if supported" in camera settings page](https://github.com/alienatedsec/yi-hack-v5/pull/299/commits/3cedddd5ab5d40d8ea432a1ae7869b4ed6a3f541)
- [Add support for native time osd](https://github.com/alienatedsec/yi-hack-v5/pull/299/commits/b305a15a5cf5b99cec55f03058e7d8635c731232)

Changes by @alienatedsec 
- [[Change] launch.sh optimised to kill gently](https://github.com/alienatedsec/yi-hack-v5/commit/63213ca4b62245b752d05ecb25d957f983c986f0) 
- [[Change] Tuning launch.sh](https://github.com/alienatedsec/yi-hack-v5/commit/26009c1e3c4e2fbe635ea7f601a1b7776c7e85c9) 
- [[New] Compile specific module - thanks](https://github.com/alienatedsec/yi-hack-v5/commit/db856f3dbde9e091cd52cf74411d363c39203347) @thelittlefireman

**Full Changelog**: https://github.com/alienatedsec/yi-hack-v5/compare/0.4.0a...0.4.1b

## What's Changed in `0.4.1a`
* Roleo testing merge for pre-release by @alienatedsec in https://github.com/alienatedsec/yi-hack-v5/pull/289

Changes by @roleoroleo 
* [Restore feature to edit config files](https://github.com/alienatedsec/yi-hack-v5/pull/289/commits/9a8871666fe542898cce1cc3c0dd5d5e5dd36e6f)
* [Add flag to decrease binary size](https://github.com/alienatedsec/yi-hack-v5/pull/289/commits/5c0e1f6d39f53edc84593ff1727380437532e75b)
* [Remove old patch](https://github.com/alienatedsec/yi-hack-v5/pull/289/commits/381b840bf043e8c02331fb0d95ff4e19e679093e)
* [Decrease RTP packet length](https://github.com/alienatedsec/yi-hack-v5/pull/289/commits/f5e97ce30d01cf6b41081c7f2a4ca3741f19bab9)
* [Improve sync/async read](https://github.com/alienatedsec/yi-hack-v5/pull/289/commits/9897afc09df6e08870401341e53262bc5a3be558)
* [Update gitignore](https://github.com/alienatedsec/yi-hack-v5/pull/289/commits/d29169e3cceb7765e9ffb182a3663b3ec51bfc0b)
* [New features: disable, force low and already running](https://github.com/alienatedsec/yi-hack-v5/pull/289/commits/70a78bd9ba275af44f93c4d11928b92af4a6d87c)
* [Remove unused settings from config files](https://github.com/alienatedsec/yi-hack-v5/pull/289/commits/d19971d4933f1addc75192ad786e3a120ba02729)
* [Add options to disable snapshot and force low res](https://github.com/alienatedsec/yi-hack-v5/pull/289/commits/c7034e24c88b105183cdb1b802ec9d4445b6d963)
* [Remove unused mqtt options](https://github.com/alienatedsec/yi-hack-v5/pull/289/commits/fcb3bc313482d30c3293fb8a5830f678ba4ba565)
* [Fix and improve h264grabber](https://github.com/alienatedsec/yi-hack-v5/pull/289/commits/2dc4cfa895413867f7bdc3b71e594c322d47401c)
* [Fix system.conf parameters](https://github.com/alienatedsec/yi-hack-v5/pull/289/commits/6ce47e243809f55bf4df60f9f5b5eadec930b2ad)
* [Remove ai_human_detection and restore baby_crying](https://github.com/alienatedsec/yi-hack-v5/pull/289/commits/a309b2aac774f2cb39f34969d7721f9c6df2a02a)

Changes by @alienatedsec 
* [Wi-Fi Recovery Partition](https://github.com/alienatedsec/yi-hack-v5/commit/25db09f312d3f05f5ed64c81ec9f4ad5a5167f64) - Instructions to be issued soon.

**Full Changelog**: https://github.com/alienatedsec/yi-hack-v5/compare/0.4.0...0.4.1a

<hr /><em>This discussion was created from the release <a href='https://github.com/alienatedsec/yi-hack-v5/releases/tag/0.4.1d'>0.4.1d</a>.</em>

**Comment by reef-actor** [comment]
I have upgraded to this version from 0.4.0 and at irregular intervals frigate would eat all the RAM in seconds and grind my server to a halt for minutes.
From the frigate logs I see these messages come up each time:
```
[2024-01-20 20:34:21] frigate.video                  ERROR   : drive: ffmpeg process is not running. exiting capture thread...
[2024-01-20 20:34:31] watchdog.drive                 ERROR   : Ffmpeg process crashed unexpectedly for drive.
[2024-01-20 20:34:31] watchdog.drive                 ERROR   : The following ffmpeg logs include the last 100 lines prior to exit.
[2024-01-20 20:34:31] ffmpeg.drive.detect            ERROR   : [rtsp @ 0x55e03ad82e40] RTP: PT=60: bad cseq b877 expected=d533
[2024-01-20 20:34:31] ffmpeg.drive.detect            ERROR   : [h264 @ 0x55e03ad98cc0] Increasing reorder buffer to 1
[2024-01-20 20:34:31] ffmpeg.drive.detect            ERROR   : [out_0_0 @ 0x55e03aea5b00] 100 buffers queued in out_0_0, something may be wrong.
[2024-01-20 20:34:31] ffmpeg.drive.detect            ERROR   : [out_0_0 @ 0x55e03aea5b00] 1000 buffers queued in out_0_0, something may be wrong.
[2024-01-20 20:44:31] watchdog.drive                 INFO    : drive exceeded fps limit. Exiting ffmpeg...
```
This seems to suggest there is corruption in the stream data I think?
It is running on a Outdoor 9CUS, base 0.3.8. I have audio disabled and am using the high res stream.
Is there anything I can do to collect more data/logging from the camera to try and work out what is happening?
